framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3592,6 +3592,29 @@ protected void checkEnclosingExpr(NewClassTree node, AnnotatedExecutableType con
                 argumentReceiverType = atypeFactory.getAnnotatedType(node.getEnclosingExpression());
             } else {
                 argumentReceiverType = atypeFactory.getReceiverType(node);
+                /*
```

Comment:
Is there a reason to do this here instead of in `getReceiverType`?
Wouldn't other callers of `getReceiverType` also benefit from this transformation?

Reply:
Thanks! I remember we couldn't do it. Let me try and explain.